src/tests/ftest/control/daos_admin_privileged.py

```diff
@@ -0,0 +1,111 @@
+    """Test Class Description:
+    Test to verify that daos_server when run as normal user, can perform
+    privileged functions.
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DaosAdminPrivTest object."""
+        super(DaosAdminPrivTest, self).__init__(*args, **kwargs)
+        self.setup_start_agents = False
+        self.setup_start_servers = False
+
+    def test_daos_admin_scm_format(self):
+        """
+        JIRA ID: DAOS-2895
+        Test Description: Test daso_admin functionality to perform format
+        privileged operations while daos_server is run as normal user.
+        :avocado: tags=all,tiny,pr,daos_admin,basic
+        """
+        # Verify that daos_admin has the correct permissions
+        self.log.info("Checking daos_admin binary permissions")
+        file_stats = os.stat("/usr/bin/daos_admin")
+        file_perms = oct(file_stats.st_mode)[-4:]
+        if file_perms != '4755':
+            self.fail("Incorrect daos_admin permissions: {}".format(file_perms))
+        
+        # Start server as non-root
+        self.log.info("Starting server as non-root")
+        server = ServerManager(self.bin, os.path.join(self.ompi_prefix, "bin"))
+        server.get_params(self)
+        server.hosts = (
+            self.hostlist_servers, self.workdir, self.hostfile_servers_slots)
+
+        if self.prefix != "/usr":
+            if server.runner.export.value is None:
+                server.runner.export.value = []
+            server.runner.export.value.extend(["PATH"])
+
+        yamlfile = os.path.join(self.tmp, "daos_avocado_test.yaml")
+        server.runner.job.set_config(yamlfile)
+        server.server_clean()
+
+        # Prep server for format, run command under non-root user
+        self.log.info("Performing SCM storage prepare")
+        user = getpass.getuser()
+        try:
+            storage_prepare(self.hostlist_servers, user, "dcpm")
+        except ServerFailed as err:
+            self.fail("Failed preparing SCM as non-root user: {}".format(err))
+
+        # Start server
+        try:
+            server.run()
+        except CommandFailure as err:
+            # Kill the subprocess, anything that might have started
+            server.kill()
+            self.fail("Failed to start server under non-root user".format(err))
+
+        # Update hostlist value for dmg command
+        port = self.params.get("port", "/run/server_config/*")
+        h_ports = [
+            "{}:{}".format(host, port) for host in self.hostlist_servers]
+
+        # Run format command under non-root user
+        self.log.info("Performing SCM format")
+        format_res = storage_format(os.path.join(self.prefix, "bin"), h_ports)
+        if format_res == None:
+            self.fail("Failed to format storage")
+
+        # Stop server
+        try:
+            server.stop()
+        except ServerFailed as error:
+            self.fail("Failed to stop server")
```

Comment:
(pylint-missing-final-newline) Final newline missing
(pylint-mixed-line-endings) Mixed line endings LF and CRLF